react-popup/my-app/src/Popup.tsx

```diff
@@ -0,0 +1,40 @@
+import { ReactNode } from 'react';
+import './Popup.css';
+import { createPortal } from 'react-dom';
+
+type Props = {
+  isOpen: boolean;
+  positionTo: HTMLElement | null;
+  onClose: () => void;
+  children: ReactNode;
+  opacity?: number;
+};
+
+export function Popup({
+  isOpen,
+  positionTo,
+  onClose,
+  children,
+  opacity = 0.5,
+}: Props) {
+  if (!isOpen) {
+    return null;
+  } else {
```

Comment:
Since you return on line 21, the `else` isn't necessary here. In my opinion, it makes the code nicer if you do not use `else` because you eliminate a level of indenting.